### PR TITLE
Fix PR handling

### PR DESCRIPTION
**Pull request by SuperQ**
* Fix the `head` search for PRs to use the org name, not the repo name.
* Only check open PRs.
* Tempalte ORG and REPO into vars to avoid typos.
* Add github_api function to handle API calls.
* Make branch name a ENV param.
* Auto-detect default branch.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

File: scripts/bump_version.sh [hunk 1]
```diff
@@ -18,11 +21,17 @@ if ! GO111MODULE=on go run ./cmd/builder-bumper; then
...
+  shift 1
+  curl --retry 5 --silent --fail -u "${GITHUB_USER}:${GITHUB_TOKEN}" "${url}" "$@"
+}
+
 if [[ -n "$(git status --porcelain)" ]]; then
   # Check if a PR is already opened for the branch.
-  prLink=$(curl --show-error --silent \
-    -u "${GITHUB_USER}:${GITHUB_TOKEN}" \
-    "https://api.github.com/repos/prometheus/golang-builder/pulls?head=golang-builder:${BRANCH_NAME}" | jq '.[0].url')
+  prLink=$(github_api "repos/${ORG}/${REPO}/pulls?state=open&head=${ORG}:${BRANCH_NAME}" | jq '.[0].html_url')
```
Comment (roidelapluie) [inline scripts/bump_version.sh:34]: why the change to html_url?

**Comment by roidelapluie** [comment]
While you are at it, can you make the branch a parameter? so we can change to main later on easily.

File: scripts/bump_version.sh (hunk 1 above)
Comment (SuperQ) [reply scripts/bump_version.sh:34]: `url` returns the `api.github.com` URL. I copied this from [Prometheus sync repo](https://github.com/prometheus/prometheus/blob/022c91c22194655ecdc506d5b1aee55bb7b0e1e0/scripts/sync_repo_files.sh#L208).

**Comment by SuperQ** [comment]
Done, made branch a param.

File: scripts/bump_version.sh [hunk 2]
```diff
@@ -37,13 +46,11 @@ if [[ -n "$(git status --porcelain)" ]]; then
...
-  if git push --quiet "https://${GITHUB_TOKEN}:@github.com/prometheus/golang-builder" --set-upstream "${BRANCH_NAME}" 1>/dev/null 2>&1; then
-    curl --show-error --silent \
-      -u "${GITHUB_USER}:${GITHUB_TOKEN}" \
+  git push --quiet "https://${GITHUB_TOKEN}:@github.com/${ORG}/${REPO}" ":${BRANCH_NAME}" 1>/dev/null 2>&1
+  if git push --quiet "https://${GITHUB_TOKEN}:@github.com/${ORG}/${REPO}" --set-upstream "${BRANCH_NAME}" 1>/dev/null 2>&1; then
+    github_api "repos/${ORG}/${REPO}/pulls" \
       -X POST \
-      -d '{"title":"Bump Go version","base":"master","head":"'${BRANCH_NAME}'","body":""}' \
-      "https://api.github.com/repos/prometheus/golang-builder/pulls"
+      -d '{"title":"Bump Go version","base":"master","head":"'${BRANCH_NAME}'","body":""}'
```
Comment (roidelapluie) [inline scripts/bump_version.sh:53]: I mean this `base` branch.

Comment (SuperQ) [reply scripts/bump_version.sh]: Ohhh, right, fixed.

**Review by roidelapluie — APPROVED** [review]
